dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/OpenTelemetryInfoFactory.java

```diff
@@ -0,0 +1,358 @@
+            return new OpenTelemetryInfo(false, OpenTelemetry.noop());
+        }
+
+        try {
+            if (AgentDetection.isAgentActive()) {
+                // If we're using the agent, it will have set GlobalOpenTelemetry and we must use its instance
+                // all config is handled by the agent in this case
+                return new OpenTelemetryInfo(true, GlobalOpenTelemetry.get());
+            }
+
+            final Map<String, String> telemetryProperties = getTelemetryProperties();
+
+            //Builds tracer provider if user has enabled tracing aspects with config properties
+            if (!checkDisabled(telemetryProperties)) {
+                OpenTelemetry openTelemetry = AccessController.doPrivileged( (PrivilegedAction<OpenTelemetry>) () -> {
+                    //This contains API calls that change between the upstream open telemetry version.
+                    //We use @Reference because the callers coming into  OpenTelemetryInfoFactory via
+                    //The static accessor do not know which version of mpTelemetry will be in use.
+                    BundleContext bc = getBundleContext(OpenTelemetryInfoFactory.class);
+                    OpenTelemetrySdkBuilderSupplier supplier = getService(bc, OpenTelemetrySdkBuilderSupplier.class);
+
+                    return supplier
+                                    .getPartiallyConfiguredOpenTelemetrySDKBuilder()
+                                    .addPropertiesCustomizer(x -> telemetryProperties) //Overrides OpenTelemetry's property order
+                                    .addResourceCustomizer(OpenTelemetryInfoFactory::customizeResource) //Defaults service name to application name
+                                    .setServiceClassLoader(Thread.currentThread().getContextClassLoader())
+                                    .build()
+                                    .getOpenTelemetrySdk();
+                });
+
+                if (openTelemetry != null) {
+                    return new OpenTelemetryInfo(true, openTelemetry);
+                }
+            }
+            //By default, MicroProfile Telemetry tracing is off.
+            //The absence of an installed SDK is a “no-op” API
+            //Operations on a Tracer, or on Spans have no side effects and do nothing
+            ComponentMetaData cData = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData();
+            String applicationName = cData.getJ2EEName().getApplication();
+            Tr.info(tc, "CWMOT5100.tracing.is.disabled", applicationName);
+
+            return new OpenTelemetryInfo(false, OpenTelemetry.noop());
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
+            return new OpenTelemetryInfo(false, OpenTelemetry.noop());
+        }
+
+    }
+
+
+    public synchronized void disposeOpenTelemetry(J2EEName j2EEName) {
```

Comment:
Don't think this needs to be `synchronized`, we only call it from the dispose method and there shouldn't be a problem calling it for two applications at the same time since `appToInfo` is a concurrent map.

Reply:
Done